server/src/test/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexServiceTests.java

```diff
@@ -900,6 +900,15 @@ public void testGetIndexNumberOfRoutingShardsWhenExplicitlyConfigured() {
         assertThat(targetRoutingNumberOfShards, is(9));
     }
 
+    public void testGetIndexNumberOfRoutingShardsWhenNullConfigured() {
```

Comment:
Does it make sense to test that this the same as when the setting is absent? `testGetIndexNumberOfRoutingShardsWithNullSourceIndex` is _similar_ but since the `INDEX_NUMBER_OF_SHARDS_SETTING` is different than here it's not **the same**.

Reply:
Yes, I can adjust the the test to also test that the the same value is set for routing shards value if the setting is absent.

Reply:
I've pushed: 7366dc647cece13e2a40b689e9a6b5a7df2ac813